### PR TITLE
First draft at standards compliant DID document

### PR DESCRIPTION
**Pull request by pelle**

### CODE REVIEW

File: pki/diddocument.md [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  "@context": "https://w3id.org/did/v1",
+  "id": "did:uport:2nQtiQG6Cgm1GYTBaaKAgr76uY7iSexUkqX",
+  "publicKey": [{
+    "id": "did:uport:2nQtiQG6Cgm1GYTBaaKAgr76uY7iSexUkqX#keys-1",
+    "type": "EdDsaSAPublicKeySecp256k1",
+    "owner": "did:uport:2nQtiQG6Cgm1GYTBaaKAgr76uY7iSexUkqX",
+    "publicKeyHex": "04613bb3a4874d27032618f020614c21cbe4c4e4781687525f6674089f9bd3d6c7f6eb13569053d31715a3ba32e0b791b97922af6387f087d6b5548c06944ab062"
+  }, {
+    "id": "did:uport:2nQtiQG6Cgm1GYTBaaKAgr76uY7iSexUkqX#keys-2",
+    "type": "Ed25519SigningKey",
```
Comment (pelle) [inline pki/diddocument.md]: @christianlundkvist @oed is this correct or do we need a new type here?

File: pki/diddocument.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+{
+  "@context": "https://w3id.org/did/v1",
+  "id": "did:uport:2nQtiQG6Cgm1GYTBaaKAgr76uY7iSexUkqX",
+  "publicKey": [{
+    "id": "did:uport:2nQtiQG6Cgm1GYTBaaKAgr76uY7iSexUkqX#keys-1",
+    "type": "EdDsaSAPublicKeySecp256k1",
+    "owner": "did:uport:2nQtiQG6Cgm1GYTBaaKAgr76uY7iSexUkqX",
+    "publicKeyHex": "04613bb3a4874d27032618f020614c21cbe4c4e4781687525f6674089f9bd3d6c7f6eb13569053d31715a3ba32e0b791b97922af6387f087d6b5548c06944ab062"
+  }],
+  "owner": {
```
Comment (pelle) [inline pki/diddocument.md]: Where should we put profile info for apps and organization? I temporarily placed it under `owner` but I'm open to suggestions.

File: pki/diddocument.md (hunk 1 above)
Comment (christianlundkvist) [reply pki/diddocument.md]: @pelle We would need something like `secp256k1SigningKey`. Unless we really mean that we are using an Ed25519 key which is certainly possible.

File: pki/diddocument.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+This is a example of a identity document for an app with extra public profile information:
+
+```js
+{
+  "@context": "https://w3id.org/did/v1",
+  "id": "did:uport:2nQtiQG6Cgm1GYTBaaKAgr76uY7iSexUkqX",
+  "publicKey": [{
+    "id": "did:uport:2nQtiQG6Cgm1GYTBaaKAgr76uY7iSexUkqX#keys-1",
+    "type": "EdDsaSAPublicKeySecp256k1",
```
Comment (christianlundkvist) [inline pki/diddocument.md]: I've already brought this up with the DID spec folks that this (`EdDsaSAPublicKeySecp256k1 `) is a weird misspelling. So we'll need something different here, like `EcdsaPublicKeySecp256k1`.

File: pki/diddocument.md (hunk 2 above)
Comment (christianlundkvist) [reply pki/diddocument.md]: @pelle I don't think "owner" should be a high-level field. Profile info stuff should be in a uport-specific field. During the calls we discussed having method-specific fields be prefixed with the method name. So in our case we could have something like `uportProfile` as the field name.

File: pki/diddocument.md [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+This is an example of a minimal identity document:
+
+```js
+{
+  "@context": "https://w3id.org/did/v1",
+  "id": "did:uport:2nQtiQG6Cgm1GYTBaaKAgr76uY7iSexUkqX",
+  "publicKey": [{
+    "id": "did:uport:2nQtiQG6Cgm1GYTBaaKAgr76uY7iSexUkqX#keys-1",
+    "type": "EdDsaSAPublicKeySecp256k1",
```
Comment (christianlundkvist) [inline pki/diddocument.md]: I've already brought this up with the DID spec folks that this (`EdDsaSAPublicKeySecp256k1`) is a weird misspelling. So should put something different here, like `EcdsaPublicKeySecp256k1`.

File: pki/diddocument.md (hunk 1 above)
Comment (pelle) [reply pki/diddocument.md]: Really this is for the encryption key which is ed25519 afaik @oed ?

File: pki/diddocument.md (hunk 2 above)
Comment (pelle) [reply pki/diddocument.md]: Great idea

File: pki/diddocument.md (hunk 4 above)
Comment (pelle) [reply pki/diddocument.md]: 👍

File: pki/diddocument.md (hunk 1 above)
Comment (christianlundkvist) [reply pki/diddocument.md]: @pelle If it's for encryption then we should call it `curve25519EncryptionKey`. CC @oed

File: pki/diddocument.md [hunk 5]
```diff
@@ -16,12 +16,12 @@ This is an example of a minimal identity document:
...
   "publicKey": [{
     "id": "did:uport:2nQtiQG6Cgm1GYTBaaKAgr76uY7iSexUkqX#keys-1",
-    "type": "EdDsaSAPublicKeySecp256k1",
+    "type": "EcdsaPublicKeySecp256k1",
     "owner": "did:uport:2nQtiQG6Cgm1GYTBaaKAgr76uY7iSexUkqX",
     "publicKeyHex": "04613bb3a4874d27032618f020614c21cbe4c4e4781687525f6674089f9bd3d6c7f6eb13569053d31715a3ba32e0b791b97922af6387f087d6b5548c06944ab062"
   }, {
     "id": "did:uport:2nQtiQG6Cgm1GYTBaaKAgr76uY7iSexUkqX#keys-2",
-    "type": "Ed25519SigningKey",
+    "type": "Ed25519EncryptionPublicKey",
```
Comment (christianlundkvist) [inline pki/diddocument.md]: It's `Curve25519` (a Montgomery curve) for encryption and Ed25519 (an Edwards curve) for signing. The curves are birationally equivalent but strictly speaking different curves. So I would call this `Curve25519EncryptionPublicKey`.

Comment (pelle) [reply pki/diddocument.md]: Gracias

File: pki/diddocument.md [hunk 6]
```diff
@@ -0,0 +1,69 @@
+# Decentralized Identity Document (DID Document)
+
+This is the subset of the [DID Document](https://w3c-ccg.github.io/did-spec) spec that we implement as part of the uPort platform.
+
+The Identity document is stored on IPFS and tied to the address using the uport registry as specified in the [PKI document](../index).
```
Comment (oed) [inline pki/diddocument.md]: Link to PKI docuement seems broken

File: pki/diddocument.md [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+    "owner": "did:uport:2nQtiQG6Cgm1GYTBaaKAgr76uY7iSexUkqX",
+    "publicKeyHex": "04613bb3a4874d27032618f020614c21cbe4c4e4781687525f6674089f9bd3d6c7f6eb13569053d31715a3ba32e0b791b97922af6387f087d6b5548c06944ab062"
+  }, {
+    "id": "did:uport:2nQtiQG6Cgm1GYTBaaKAgr76uY7iSexUkqX#keys-2",
+    "type": "Curve25519EncryptionPublicKey",
+    "owner": "did:uport:2nQtiQG6Cgm1GYTBaaKAgr76uY7iSexUkqX",
+    "publicKeyBase64": "QCFPBLm5pwmuTOu+haxv0+Vpmr6Rrz/DEEvbcjktQnQ="
+  }],
+  "authentication": [{
+    "type": "EcdsaSignatureAuthentication2018",
```
Comment (oed) [inline pki/diddocument.md]: Why `2018`?

File: pki/diddocument.md [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+
+```js
+{
+  "@context": "https://w3id.org/did/v1",
+  "id": "did:uport:2nQtiQG6Cgm1GYTBaaKAgr76uY7iSexUkqX",
+  "publicKey": [{
+    "id": "did:uport:2nQtiQG6Cgm1GYTBaaKAgr76uY7iSexUkqX#keys-1",
+    "type": "EcdsaPublicKeySecp256k1",
+    "owner": "did:uport:2nQtiQG6Cgm1GYTBaaKAgr76uY7iSexUkqX",
+    "publicKeyHex": "04613bb3a4874d27032618f020614c21cbe4c4e4781687525f6674089f9bd3d6c7f6eb13569053d31715a3ba32e0b791b97922af6387f087d6b5548c06944ab062"
```
Comment (oed) [inline pki/diddocument.md]: Should we also add the public encryption key?

Comment (pelle) [reply pki/diddocument.md]: @oed yes one of the examples has that

File: pki/diddocument.md (hunk 7 above)
Comment (pelle) [reply pki/diddocument.md]: That seems to be the convention. It's weird to me as well.

@christianlundkvist does this make sense to you?

Comment (christianlundkvist) [reply pki/diddocument.md]: That’s the convention that they’ve been using for these key suites, but I’m not sure where exactly these key suites are defined. There might be one for ecdsa already but that’s probably not 2018. (The year is basically a version number)

Comment (christianlundkvist) [reply pki/diddocument.md]: @pelle Found the crypto suites registry:

https://w3c-ccg.github.io/ld-cryptosuite-registry/

**Pull request merged**